pkg/plugins/datasource/wrapper/datasource_plugin_wrapper_v2.go

```diff
@@ -51,6 +53,22 @@ func (tw *DatasourcePluginWrapperV2) Query(ctx context.Context, ds *models.DataS
 		return nil, err
 	}
 
+	if query.Headers == nil {
+		query.Headers = make(map[string]string)
+	}
+
+	if ds.JsonData != nil && ds.JsonData.Get("oauthPassThru").MustBool() {
```

Comment:
I think this helper method would fit well in pkg/services/oauthtoken since both datasource_plugin_wrapper_v2.go and ds_proxy.go will be accessing the same package to get the current OAuthToken.

Reply:
Yeah makes sense. I wrote that comment before realizing that things probably should be moved around.